Framework/CurveFitting/src/Algorithms/DoublePulseFit.cpp

```diff
@@ -0,0 +1,405 @@
+  auto pulseOffset = getProperty("PulseOffset");
+  auto firstPulseWeight = getProperty("FirstPulseWeight");
+  auto secondPulseWeight = getProperty("SecondPulseWeight");
+  if (auto paramFunction =
+          std::dynamic_pointer_cast<Mantid::API::ParamFunction>(function)) {
+    doublePulseFunction = getDoublePulseFunction(
+        paramFunction, pulseOffset, firstPulseWeight, secondPulseWeight);
+  } else if (auto multiDomainFunction =
+                 std::dynamic_pointer_cast<Mantid::API::MultiDomainFunction>(
+                     function)) {
+    doublePulseFunction = getDoublePulseFunction(
+        multiDomainFunction, pulseOffset, firstPulseWeight, secondPulseWeight);
+  } else {
+    throw std::runtime_error("Incompatible function form. DoublePulseFit.cpp");
+  }
+
+  runFitAlgorith(fitAlg, doublePulseFunction, getProperty("MaxIterations"));
+
+  createOutput(fitAlg, doublePulseFunction);
+}
+
+/**
+ * This function declares the additional output properties which are not
+ * present in init. It is done this way to match the behaviour of Fit which
+ * declares many of its output properties internally setting the names of
+ * output workspaces based upon the input property Output.
+ */
+void DoublePulseFit::declareAdditionalProperties() {
+  std::string baseName = getProperty("Output");
+  if (baseName.empty()) {
+    API::Workspace_const_sptr ws = getProperty("InputWorkspace");
+    baseName = ws->getName();
+  }
+
+  Mantid::API::IFunction_sptr function = getProperty("Function");
+  if (m_makeOutput) {
+    declareProperty(
+        std::make_unique<API::WorkspaceProperty<API::ITableWorkspace>>(
+            "OutputNormalisedCovarianceMatrix", "", Kernel::Direction::Output),
+        "The name of the TableWorkspace in which to store the final "
+        "covariance "
+        "matrix");
+    setPropertyValue("OutputNormalisedCovarianceMatrix",
+                     baseName + "NormalisedCovarianceMatrix");
+
+    declareProperty(
+        std::make_unique<API::WorkspaceProperty<API::ITableWorkspace>>(
+            "OutputParameters", "", Kernel::Direction::Output),
+        "The name of the TableWorkspace in which to store the "
+        "final fit parameters");
+
+    setPropertyValue("OutputParameters", baseName + "Parameters");
+
+    if (m_outputFitData) {
+      if (m_multiDomain) {
+        declareProperty(
+            std::make_unique<
+                API::WorkspaceProperty<Mantid::API::WorkspaceGroup>>(
+                "OutputWorkspace", "", Kernel::Direction::Output),
+            "Name of the output Workspace holding resulting simulated "
+            "spectrum");
+        setPropertyValue("OutputWorkspace", baseName + "Workspace");
+      } else {
+        declareProperty(
+            std::make_unique<
+                API::WorkspaceProperty<Mantid::API::MatrixWorkspace>>(
+                "OutputWorkspace", "", Kernel::Direction::Output),
+            "Name of the output Workspace holding resulting simulated "
```

Comment:
do we mean simulated or fitted?

Reply:
I think simulated is used here because the workspace contains the spectrum created by applying the fitted function to the range. Whilst I agree that fitted might be the clearer way to say this the current wording is the same as that used in the Fit algorithm and I would like to retain the consistency here.

Reply:
can we put fitted in brackets after simulated? It just makes me uneasy as it is a bit misleading and we are starting to see more simulated data.

Reply:
I'll just change to fitted.